Fixes an error displaying in the module list (the error occurs if there are disabled extensions)

astridx opened the pull request:
If my solution will be applied, this is also a solution for the issue
https://github.com/joomla/joomla-cms/issues/7115
# See the failure
1. Go to the Extensions | Module Manager and ensure, that you have at least two modules of the type search for the location site. One is published and another is unpublished.

![snap1](https://cloud.githubusercontent.com/assets/9974686/8267657/a8ad071c-176a-11e5-8dc1-732829033a22.jpg)
1. Switch to Extension Manager|Manage and disable the Modul “Search” for the location site.

![snap2](https://cloud.githubusercontent.com/assets/9974686/8267661/c4613f1e-176a-11e5-8236-ecb036fcc338.jpg)
1. Go back to Extensions | Module Manager and see, that the Status show wrong values. The published Module looks like unpublished and the unpublished value looks like trashed.

![snap3](https://cloud.githubusercontent.com/assets/9974686/8267666/e00a8036-176a-11e5-8f4e-f50e4b61c244.jpg)
# Apply my patch
1. Go to the Extensions | Module Manager and ensure, that you have at least two modules of the type "Search" for the location site. One is published and another is unpublished.

![snap1](https://cloud.githubusercontent.com/assets/9974686/8267672/2fa69f9e-176b-11e5-8821-76f62f942409.jpg)
1. Switch to Extension Manager|Manage and disable the Module “Search” for the location site.

![snap2](https://cloud.githubusercontent.com/assets/9974686/8267684/71a321f6-176b-11e5-979e-cb6ee5fe0bf2.jpg)
1. Go back to Extensions | Module Manager and see, that the Status did not show the wrong values any more. The Module with disabled extensions has another icon with a message, that explained why you could not change the status in this list.

![snap3](https://cloud.githubusercontent.com/assets/9974686/8267690/a396c83e-176b-11e5-9388-ebb5905667d9.jpg)

I am pleased about any feedback

Commit:
Disable view of state in Module List when the extension is disabled. Before the state was shown wrong.

Commit:
Disable view of state in Module List when the extension is disabled. Before the state was shown wrong.

zero-24 committed:
Update default.php

Commit:
Merge pull request #1 from zero-24/patch-2

Thank you :-)

Commit:
The first attempt to improve my proposal :-)

Add an icon with a tooltip explaining that the extension has been
disabled.

Bakual commented on administrator/components/com_modules/views/modules/tmpl/default.php:
```diff
@@ -161,21 +161,26 @@
+								<?php echo JHtml::_('actionsdropdown.render', $this->escape($item->title)); ?>
+							<?php else : ?>
+								<?php // Extension is not enabled, show a message that indicates this. ?>
+								<button class="btn-micro hasTooltip" data-original-title="<?php echo JText::_('COM_MODULES_MSG_MANAGE_EXTENSION_DISABLED'); ?>"><i class="icon-ban-circle"></i></button>
```
You should be able to use just the `title` attribute instead of `data-original-title`. It's more in line with how we do in other places.

Bakual commented on administrator/language/en-GB/en-GB.com_modules.ini:
```diff
@@ -87,6 +87,7 @@ COM_MODULES_MODULE_TEMPLATE_POSITION="%1$s (%2$s)"
 COM_MODULES_MODULES="Modules"
 COM_MODULES_MODULES_FILTER_SEARCH_DESC="Search in module title."
 COM_MODULES_MSG_MANAGE_NO_MODULES="There are no modules matching your query"
+COM_MODULES_MSG_MANAGE_EXTENSION_DISABLED="<p>You can't change this value because the extension of the module is disabled.</p><p>Please enable the extension. You can do this using</p> <p>Extension manager - Manage.<p>"
```
I wouldn't put so many `<p>`tags into the string. It doesn't look very well. Only use a `<br />` if a linebreak is really needed.

Bakual commented:
I like that solution. Very well done, thanks already!

I added some comments.

Bakual commented:
Btw: Test was successful.

Commit:
removed the <p>-Tags in language string

Commit:
changed a title Attribut

brianteeman commented:
In the tooltip it is enough to say. This module is disabled. Use Extension Manager... ... to enable it.

We don't usually have any p markup in a tooltip and with a simpler text you probably don't even need a line break.

astridx commented:
Thank you for the feedback and the test.
I applied Bakuals suggestions.
@brianteeman
Are you sure, that the module is disabled? My point of view is this, that the module has the state published. Only the extension is (perhaps by mistake) disabled.
<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/7221">issues.joomla.org/joomla-cms/7221</a>.</sub>

brianteeman commented:
In your instructions you state to disable the module <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/7221">issues.joomla.org/joomla-cms/7221</a>.</sub>

Commit:
Update en-GB.com_modules.ini

astridx commented:
@brianteeman
Yes, you are right; 

Now (and in my subconscious perhaps even earlier, because I stated it in my one text) that I have been thinking about it more precisely, I changed my opinion. Now I agree that a module is an extension (Just as a horse is an animal ….).

I've changed my text in your text :-).

Thanks for your note.

brianteeman commented:
Language string looks good now

brianteeman commented:
Tested as per instructions and looks good to me and resolves a silly bug (note before this patch you can click on publish for the module and it will say `1 module successfully published.` but will stay unpublished because the module is disabled<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/7221">issues.joomla.org/joomla-cms/7221</a>.</sub>

zero-24 commented:
Thanks :smile: RTC!<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/7221">issues.joomla.org/joomla-cms/7221</a>.</sub>

Commit:
Backend - list views: Alignment fixes for almost all core components (similar to #7384) Fixes #7421

brianteeman committed:
NOTE

Commit:
Properly allow "0" value for published/state field

shur committed:
[cs] Add dumping data comment

Commit:
Moving association saving to JModelAdmin and using $associationsContext property.

n9iels committed:
Add microdata to breadcrumbs Fixes #7613

Commit:
Tag edit form cleanup Fixes #7628

brianteeman committed:
Update

Commit:
Merge pull request #7469 from brianteeman/acl-rules

Simplify text and Elminitate Redundancy for JLIB_RULES_SETTING_NOTES

rdeutz committed:
Make JToolbarHelper::custom calls consistent

# Executive summary
In some part of the code the JToolbarHelper::custom method was called with 6 parameters but the method has only 5 parameters.

# Backwards compatibility
Full b/c, no problems are expected

# Translation impact
none

# Testing instructions
Because the parameter was ignored before there isn’t so much to test. Just make sure the following backend views are still working and the buttons are doing what they did before

I have changed the custom Buttons in this views:

* Extensions/Manage/Update ->Update, Find Updates, Clear Cache
* Extensions/Manage/Discover -> Install, Discover
* Extensions/Manage/Database -> Fix
* Extensions/Manage/Install Languages -> Install, Find Languages
* Components/Joomla! Update -> Clear Cache
* Extensions/Templates -> open a Template:
** Select an Image -> Crop
** Select a Zip Archive (maybe copy a zip into the template Dir) -> Extract Here
** Select a less file -> Compile Less

Thanks to @brianteeman for finding the inconsistency

brianteeman committed:
fix error in ini

Commit:
Merge pull request #7634 from brianteeman/acl-rules-fix

fix error in ini #7469

rdeutz committed:
Merge pull request #7617 from Bakual/SaveAssoc

Moving association saving to JModelAdmin

rdeutz committed:
Merge pull request #7629 from infograf768/protostar_collapse

Long drop-down sub menu gets cropped in Mobile View (solves #7587)

wilsonge committed:
Merge pull request #7538 from shur/patch-8

[cs] Add dumping data comment

Commit:
Merge pull request #7630 from rdeutz/jtoolbar_custom_call

Make JToolbarHelper::custom calls consistent

Kubik-Rubik committed:
Merge pull request #7488 from Bakual/FixDefaultAccessFix

Fixing default values for article/module/category creation

Commit:
erste Installation

erste Installation

zero-24 commented:
@astridx any reason to close / delete the branch?

astridx commented:
Sorry :(

zero-24 commented:
No Problem RTC'ing again :smile: <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/7221">issues.joomla.org/joomla-cms/7221</a>.</sub>

infograf768 commented:
This should be changed I guess:
` +COM_MODULES_MSG_MANAGE_EXTENSION_DISABLED="This module is disabled. Use Extension Manager | Manage to enable it." `
as the wording "manager" has been taken off from en-GB in a recent PR

So, maybe something like
` +COM_MODULES_MSG_MANAGE_EXTENSION_DISABLED="This module is disabled. Use Extension => Manage to enable it."

zero-24 committed:
Remove the wording manager impments @infograf768 suggestion

zero-24 commented:
@astridx see: https://github.com/astridx/joomla-cms/pull/2 that implements the tip by @infograf768 Thanks :smile:

Commit:
Merge pull request #2 from zero-24/patch-7

Remove the wording manager impments @infograf768 suggestion

infograf768 commented:
Oops: should be "Extensions => Manage" and not "Extension => Manage"

Commit:
Typo corrected

Extensions => Manage 
and not 
Extension => Manage

Commit:
Merge branch 'staging' of https://github.com/astridx/joomla-cms into ModuleList

joommarkt commented:
Sorry, I made a mistake. How can withdraw the last commit?

Bakual commented:
Looks like a wrong merge commit. PR is currently messed up a lot.

Removing RTC.

astridx commented:
Sorry, I tided up … I will learn some more about git before I try a next PR :-(
Is it possible to undo the last merge? I do not want to make another failure.....

Bakual commented:
In Git everything is possible :smile: 
You can revert the commit or check out an earlier one and force-push the branch. Both would solve it.

zero-24 commented:
@astridx see:
https://github.com/blog/2019-how-to-undo-almost-anything-with-git

```
git revert 7e731a89f1043e9c015d47ea2817064b3b057b36
git commit -am 'try to revert the bad merge'
git push origin ModuleList -f 
```

Should do the trick and revert the commit that breaks everything ;)

other way arround (https://www.atlassian.com/git/tutorials/viewing-old-commits/):

```
git checkout 5abbe02be69579b9bcbc8a58b78a3c0bdccb090e
git commit -am 'try to revert the bad merge'
git push origin ModuleList -f 
```

If that don't work we can redo all changes that can be shown here:
https://github.com/joomla/joomla-cms/compare/staging...astridx:5abbe02be69579b9bcbc8a58b78a3c0bdccb090e

If you need more info about git in general see:
https://git-scm.com/book/en/v2
http://phproberto.com/en/32-git-github-basic-workflow
https://www.youtube.com/watch?v=FsNjqy-Q4mc

astridx commented:
Sorry, but it was not able to revert the merge. It ends all the time
that with the error fatal: bad object
7e731a89f1043e9c015d47ea2817064b3b057b36.
What does it mean: “If that don't work we can redo all changes that can
be shown here:” Do you redo the whole PR or only my bad merge?

zero-24 commented:
Redo the pr is the easyest. I have no git client on my phone 😉

astridx commented:
OK, then you delete this PR. Should I try to make the new PR?

zero-24 commented:
Yes you can do a new PR that based on the current staging and close this here.

Thanks.

astridx commented:
New PR: Fix a failure in the Module List #8020
